crates/staged-sync/src/utils/init.rs

```diff
@@ -7,18 +11,41 @@ use reth_db::{
 };
 use reth_primitives::{stage::StageId, Account, Bytecode, ChainSpec, H256, U256};
 use reth_provider::{DatabaseProviderRW, PostState, ProviderFactory, TransactionError};
-use std::{path::Path, sync::Arc};
+use std::{fs, path::Path, sync::Arc};
 use tracing::debug;
 
 /// Opens up an existing database or creates a new one at the specified path.
 pub fn init_db<P: AsRef<Path>>(path: P) -> eyre::Result<Env<WriteMap>> {
-    std::fs::create_dir_all(path.as_ref())?;
+    if is_db_empty(&path) {
+        fs::create_dir_all(&path)?;
+        create_db_version_file(&path)?;
+    } else {
+        check_db_version_file(&path)?;
+    }
+    update_client_version_file(&path)?;
+
     let db = Env::<WriteMap>::open(path.as_ref(), reth_db::mdbx::EnvKind::RW)?;
+
     db.create_tables()?;
 
     Ok(db)
 }
 
+/// Check if a db is empty. It does not provide any information on the
+/// validity of the data in it.
+/// We consider a database as non empty when it's a non empty directory.
+fn is_db_empty<P: AsRef<Path>>(path: P) -> bool {
```

Comment:
Would like to move away from these utils and towards methods on a database struct wrapper, which abstracts away all mdbx/path-related internals. Would also make this nicer with DB::is_empty/open/init